src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from "react";
+import { useNotesContext } from "../../utils/contexts/NotesContext";
+import ColorOptions from "../ColorOptions";
+import "./Modal.styles.css";
+function Modal(props) {
+  const { note, closeEditing } = props;
+  const { notes, saveNotes } = useNotesContext();
+  const [values, setValues] = useState(note);
+  const [openMenu, setOpenMenu] = useState(false);
+
+
+  const { title, text, color } = values;
+
+  const editNote = () =>{
+    const newList = [...notes];
+    const edited = newList.find((f) => f.id === note.id);
+    edited.title = values.title;
+    edited.text = values.text;
+    edited.color = values.color;
+    saveNotes(newList);
```

Comment:
an alternative to this code is:

```jsx
const oldNote = notes.find(({ id }) => id === note.id);
const updatedNote = {
  ...oldNote,
  ...values,
}
const newList = [
  ...notes.filter(({ id }) => id !== note.id),
  updatedNote,
];

saveNotes(newList);
```

Note that it will lose the original ordering, if the ordering is important the creation date could be stored or an object could be used instead of an array